sklearn/base.py

```diff
@@ -433,6 +439,44 @@ def _validate_data(self, X, y=None, reset=True,
 
         return out
 
+    def _set_callbacks(self, callbacks):
+        """Set callbacks for the estimator.
+
+        In the case of meta-estmators, callbacks are also set recursively
+        for all child estimators.
```

Comment:
Thoughts on doing this vs letting users set callbacks on sub-estimator instances?

what about e.g. early stopping when we ultimately support this?